js/socketclient.js

```diff
@@ -8,7 +8,9 @@ var MMSocket = function(moduleName) {
 	self.moduleName = moduleName;
 
 	// Private Methods
-	self.socket = io("/" + self.moduleName);
+	self.socket = io("/" + self.moduleName, {
+		path: window.location.pathname + "/socket.io"
```

Comment:
the slash before `socket.io` causes socket-errors when the path is empty (which means = `/`). Without the slash it works for me with subpath and without.

Reply:
but what if the `pathname` does not contain a `/` at the end

Reply:
I don't code in js, but what about[ path.join](https://nodejs.org/api/path.html#path_path_join_paths)?

Reply:
here we are not in a node.js environment, this is executed in the web browser

Reply:
I did the following tests:

http://mm.localhost/ with "/socket.io"
````
Request URL:http://mm.localhost//socket.io/?EIO=3&transport=polling&t=N1UpKax
Request Method:GET
Remote Address:[::1]:80
Status Code:
404
Version:HTTP/1.1
````

http://mm.localhost/ with "socket.io"
````
Request URL:http://mm.localhost/socket.io/?EIO=3&transport=polling&t=N1Upjzt
Request Method:GET
Remote Address:[::1]:80
Status Code:
200
Version:HTTP/1.1
````

http://localhost/mm/ with "/socket.io"
````
Request URL:http://localhost/mm//socket.io/?EIO=3&transport=polling&t=N1UrITs
Request Method:GET
Remote Address:127.0.0.1:80
Status Code:
200
Version:HTTP/1.1
````

http://localhost/mm/ with "socket.io"
````
Request URL:http://localhost/mm/socket.io/?EIO=3&transport=polling&t=N1Ur1Cb
Request Method:GET
Remote Address:127.0.0.1:80
Status Code:
200
Version:HTTP/1.1
````

So the first test fails und I think this is the reason for the failed travis-tests.
The mirror fails to load module data, they remain in the state 'loading':

![grafik](https://user-images.githubusercontent.com/25914086/74866822-c6beb780-5353-11ea-8586-7e6b0bcedf2b.png)

Reply:
> but what if the `pathname` does not contain a `/` at the end

may depends on the reverse-proxy setup. In my case http://mm.localhost/ is redirected to http://mm.localhost, http://localhost/mm/ works and http://localhost/mm is not found. But this could be fixed with a rewrite rule in the proxy.

What about testing the end of the path and adding the `/` before `socket.io` only when necessary?

Reply:
I wanted to keep the changes as simple as possible. Adding additional logic to create the path is one way to solve this.

But it is also possible to add a configuration option, where you can configure the proxy URL, so other modules can use it too. This is required if some modules want to create self links, they must use the proxy URL as base. Because all path that are using localhost with absolute URLs are not accessible when using the Proxy.

Reply:
OK i did a quic test with `http://somedomain.com/magic-mirror` without `/` as proxy url, and this Proxy setup will break the complete application. Because the browser then uses `http://somedomain.com/` as base and all secondary requests fail.

Reply:
So as of this I would consider `pathname` not containing a `/` at the end is the result of a misconfigured proxy and can be ignored here.